internal/exec/stages/files/units.go

```diff
@@ -17,57 +17,145 @@ package files
 import (
 	"fmt"
 	"path/filepath"
+	"strconv"
 	"strings"
 
+	"github.com/coreos/go-systemd/dbus"
 	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
 	"github.com/coreos/ignition/v2/internal/distro"
 	"github.com/coreos/ignition/v2/internal/exec/util"
 )
 
+type Preset struct {
+	unit            string
+	enabled         bool
+	instantiateable bool
+	instances       []string
+}
+
+// isRequiredSystemdVersionExists checks the version of Systemd
+// in a given FCOS/RHCOS system.
+func (s *stage) isRequiredSystemdVersionExists() error {
+	var systemdVersion string
+	conn, err := dbus.New()
+	if err != nil {
+		return err
+	}
+	version, err := conn.GetManagerProperty("Version")
+	if err != nil {
+		return err
+	}
+	//for FCOS, the version of systemd is printed like 'v243.5-1.fc31'
+	if strings.HasPrefix(version, "v") {
+		arr := strings.Split(version, ".")
+		firsthalf := arr[0]
+		systemdVersion = firsthalf[2:]
+	} else {
+		//for RHCOS, the version of systemd is printed like '239'
+		systemdVersion = version[1 : len(version)-1]
+	}
+	if value, err := strconv.Atoi(systemdVersion); err == nil {
+		if value < 240 {
+			if err := s.Logger.Warning("The version of systemd is less than 240, Enabling/disabling instantiated units may not work. See https://github.com/coreos/ignition/issues/586 for more information."); err != nil {
```

Comment:
Make that first comma and period. 

Also for clarity let's include the value we parsed here in case we ever get parsing errors in the future.

```
The version of systemd (239) is less than 240. Enabling/disabling instantiated units may not work. See https://github.com/coreos/ignition/issues/586 for more information.
```

Reply:
Fixed.